Lib/test/test_builtin.py

```diff
@@ -1573,6 +1573,28 @@ def test_zip_pickle(self):
             z1 = zip(a, b)
             self.check_iter_pickle(z1, t, proto)
 
+    def test_zip_pickle_strict(self):
+        a = (1, 2, 3)
+        b = (4, 5, 6)
+        t = [(1, 4), (2, 5), (3, 6)]
+        for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+            z1 = zip(a, b, strict=True)
+            self.check_iter_pickle(z1, t, proto)
+
+    def test_zip_pickle_strict_fail(self):
+        a = (1, 2, 3)
+        b = (4, 5, 6, 7)
+        for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+            z1 = zip(a, b, strict=True)
+            z2 = pickle.loads(pickle.dumps(z1, proto))
+            self.assertEqual(next(z1), next(z2))
+            self.assertEqual(next(z1), next(z2))
+            self.assertEqual(next(z1), next(z2))
+            with self.assertRaises(ValueError):
+                next(z1)
+            with self.assertRaises(ValueError):
+                next(z2)
```

Comment:
Let me propose another pickle test. Let's take a zip() iterator, pickle it under Python 3.9, and save the string. Then check that it can still be unpickled correctly here. Also check that pickling the same zip() iterator produces the same string under 3.10 as it did under 3.9. (Or perhaps the latter test is sufficient, since we already know that it roundtrips.) Bonus points for doing this for all pickle protocols supported by 3.9.

Reply:
Interesting idea! I'll have that up tonight.

Reply:
Although, this *does* effectively lock in our pickle format going forward (and not just us, but also tuples / tuple iterators / ints / whatever we zip in the test. I can do an ad-hoc verification that it works, but are we sure it needs a unit test?

Reply:
Done (added the same test for strict pickles as well).